www/src/components/CustomThemesForm.tsx

```diff
@@ -0,0 +1,190 @@
+    submitForm,
+  }));
+
+  const handleAddUrl = () => {
+    append({ url: '' });
+  };
+
+  const handleRemoveUrl = (index: number) => {
+    remove(index);
+  };
+
+  return (
+    <Form id="customThemesForm" onSubmit={handleSubmit(onSubmit)}>
+      <Form.Text className="mb-3">
+        Add a custom theme name and one or more CSS URLs to apply your own theme. The CSS files should be
+        accessible via URLs and must end with <code>.css</code>.
+      </Form.Text>
+      <Form.Group controlId="customThemeName" isInvalid={!!errors.name}>
+        <Form.Label>Theme Name</Form.Label>
+        <Form.Control
+          size="sm"
+          {...register('name', {
+            required: 'Theme name is required.',
+            validate: (value) => value.trim().length > 0 || 'Theme name is required.',
+          })}
+        />
+        {errors.name && (
+          <Form.Control.Feedback type="invalid">
+            {errors.name.message}
+          </Form.Control.Feedback>
+        )}
+      </Form.Group>
+      {fields.map((field, idx) => (
+        <Form.Group
+          key={field.id}
+          controlId={`customThemeUrl${idx}`}
+          isInvalid={!!errors.urls?.[idx]}
+        >
+          {idx === 0 && (
+            <Form.Label>CSS URL</Form.Label>
+          )}
+          <Stack direction="horizontal" gap={1}>
+            <Controller
+              name={`urls.${idx}.url`}
+              control={control}
+              rules={{
+                required: 'CSS URL is required.',
+                validate: (value) => {
+                  if (!value || value.trim().length === 0) {
+                    return 'CSS URL is required?!?!.';
+                  }
+                  if (!isValidCssUrl(value.trim())) {
+                    return 'Please enter a valid CSS URL (must start with http(s) and end with .css).';
+                  }
+                  return true;
+                },
+              }}
+              render={({ field: formField }) => (
+                <Form.Control
+                  size="sm"
+                  type="url"
+                  placeholder="https://cdn.example.com/theme.css"
+                  aria-label={idx === 0 ? undefined : 'Additional CSS URL'}
+                  {...formField}
+                  ref={el => {
+                    urlInputRefs.current[idx] = el;
+                  }}
+                />
+              )}
+            />
+            {fields.length > 1 && (
+              <IconButton
+                variant="danger"
+                src={Close}
+                onClick={() => handleRemoveUrl(idx)}
+                alt="Remove URL"
+                size="inline"
+              />
+            )}
+          </Stack>
+          {errors.urls?.[idx] && (
+            <Form.Control.Feedback type="invalid">
+              {errors.urls[idx]?.url?.message}
+            </Form.Control.Feedback>
+          )}
+        </Form.Group>
+      ))}
+      <Button
+        variant="link"
```

Comment:
Using the `link` variant here feels a little odd to me, specifically because of the underlined text when hovering (even when not hovering over the text)

![image](https://github.com/user-attachments/assets/c52c3b89-9c27-4b32-bf27-6c7b7eb0e777)

Reply:
[update] Ultimately landed on using `outline-primary`:

<img width="832" alt="image" src="https://github.com/user-attachments/assets/9aed88a6-b658-4776-af3d-2a9be8e6bf7d" />